[WIP] Calendars export to other devices (issue#824)

Seal98 opened the pull request:
Creating a servlet that allows external programs to subscribe to the Sigmah calendar using URL

Seal98 committed:
[WIP] Calendars export to other devices (issue#824)


Creating a servlet for making a subscription to the URL

Seal98 committed:
[WIP] creating share buttons (issue#824)


Creating share buttons with a popup window with the URL after its
pushing

Seal98 committed:
(issue#824) Export Sigmah calendar events to iCal format.
 
Implemented: 
-Added showing share calendar events URL with parameters: 
event type and project id
-Added tool tips to share calendar buttons
-Export servlet is renamed to ExportCalendar
-Added fetching calendar events by the servlet for specified project id.
-Added generation by servlet iCal output with Sigmah calendar events.

Seal98 committed:
(issue#824)  Refactoring and code cleanup.
 
Performed refactoring and code cleanup.

Seal98 committed:
(issue#824)  Refactoring and code cleanup #2.
 
Performed refactoring and code cleanup according Codacy review remarks

Seal98 committed:
(issue#824) tooltip, chain-button


Looltip for each button was changed. The functional for the events
button works

osarrat commented:
Hi @Seal98, 

Great to see so much work on this issue #824 ! Looking forward to the demo tomorrow !
A quick feedback on commit presentation:

- no need to start your commit message with "[WIP]" : the "[WIP]" marker is only for the Pull Request title globally, not for individual commits
- important: add a space when typing "issue #824" in your commit title: typed as you did it, it won't be recognized by Mantis

See you tomorrow !

Seal98 committed:
(issue#824) Refresh in Outlook Sigmah calendar imported by URL .


Added possibility to update(refresh) in Outlook imported by URL Sigmah
calendar with defined in ExportCalendar servler time interval.
Added name to imported calendar.

Seal98 committed:
(issue#824) Getting from Request User-agent and ip info for logging.

Added methods to get from Request User-agent and ip info for logging to
see who calls ExportCalendar servlet.

Seal98 committed:
(issue#824) Method getClientIpAddr is refactored to comply Codacy check.

-Method getClientIpAddr is refactored to comply with Codacy check.
-IndexedDB fix to allow run Sigmah online is applied.

Seal98 committed:
(issue #824) share-link buttons.

Hiding all Share-link buttons except the 'events' button

Seal98 committed:
(issue #824) Calendar URL parameter values encryption\decryption.

-Implemented URL parameter values encryption when URL is created.
-Implemented URL parameter values decryption when URL is processed by
export calendar servlet.
-Made the URL look like a filename ending with ".ics"

Seal98 committed:
(issue #824)Set Time Zone, Publish, Organizer to export events .

-Added to Set to export calendar events Time Zone, Publish, Organizer.
-Performed code changes according Codacy code check.

ghost commented:
At first sight, this can't be merged without a global clean-up (altered license headers, personal settings in pom.xml, offline mode disabled...).

osarrat commented:
Thanks @numero-six for this review !
@Seal98 , do you feel you can do those necessary changes in order to merge your pull request ? Could you by the way also merge your two pull requests which both impact the calendar ?
